lib/dependency-manager-adapters/npm.js

```diff
@@ -7,13 +7,37 @@ var path = require('path');
 var extend = require('extend');
 var debug = require('debug')('ember-try:dependency-manager-adapter:npm');
 var rimraf = RSVP.denodeify(require('rimraf'));
+var execFileSync = require('child_process').execFileSync;
 
 module.exports = CoreObject.extend({
   init: function() {
     this._super.apply(this, arguments);
     this.run = this.run || require('../utils/run');
+    this.adapterInit();
   },
-  configKey: 'npm',
+  adapterInit() {
+    this.yarnLockPresent = fs.existsSync(path.join(this.cwd, this.yarnLock));
+    let yarnManagerPresent;
+    try {
+      execFileSync('yarn', [].concat(['--version']));
+      yarnManagerPresent = true;
+    } catch (e) {
+      yarnManagerPresent = false;
+    }
+    if (this.yarnLockPresent || yarnManagerPresent) {
```

Comment:
I suggested that it would be either/or ... I was thinking of the scenario where I add a `yarn.lock` to a project, then CI runs, but I neglected to install `yarn` on my CI system. I would expect ember-try to use yarn, but it uses npm and everything works, so I wouldn't realize it wasn't using yarn except by looking through all the output.  Maybe we should warn if using `npm` with a yarn.lock present?